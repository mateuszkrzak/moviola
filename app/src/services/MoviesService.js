import Axios from 'axios';

export const API_URL = 'http://react-cdp-api.herokuapp.com/movies';

async function getMovies(options) {
  const response = await Axios.get(API_URL, {
    params: {
      ...options,
      sortOrder: 'desc',
    },
  });

  return response;
}

async function getMoviesByGenres(genres) {
  const params = new URLSearchParams();
  genres.forEach(value => params.append('filter', value));

  const response = await Axios.get(API_URL, { params });

  return response;
}

async function getMovieById(id) {
  const response = await Axios.get(`${API_URL}/${id}`);

  return response;
}

export default { getMovies, getMoviesByGenres, getMovieById };
