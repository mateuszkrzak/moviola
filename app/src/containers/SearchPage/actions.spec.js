import createMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import getMoviesAsync from './actions';
import MoviesService from '../../services/MoviesService'; // eslint-disable-line

const middlewares = [thunk];
const mockStore = createMockStore(middlewares);
jest.mock('../../services/MoviesService');

describe('Search actions', () => {
  const mockMovie = {
    title: 'Test tile',
    release_date: '2020-10-20',
    vote_average: 9,
    poster_path: 'http://poster.path',
    runtime: 120,
    overview: 'Test overview',
    genres: ['Test genre', 'Another test genre'],
  };
  let store;

  describe('get movies', () => {
    it('creates FETCH_MOVIES_SUCCESS action when request is successful', async () => {
      MoviesService.getMovies.mockImplementationOnce(() => ({
        data: { data: [mockMovie, mockMovie], total: 2 },
      }));
      store = mockStore({});

      await store.dispatch(getMoviesAsync());

      expect(store.getActions()).toMatchSnapshot();
    });

    it('creates FETCH_MOVIES_FAILURE action when request returns error', async () => {
      MoviesService.getMovies.mockImplementationOnce(() => {
        throw new Error('Test error message');
      });
      store = mockStore({});

      await store.dispatch(getMoviesAsync());

      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
