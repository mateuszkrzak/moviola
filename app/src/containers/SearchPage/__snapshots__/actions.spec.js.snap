// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search actions get movies creates FETCH_MOVIES_FAILURE action when request returns error 1`] = `
Array [
  Object {
    "type": "FETCH_MOVIES_REQUEST",
  },
  Object {
    "payload": "Test error message",
    "type": "FETCH_MOVIES_FAILURE",
  },
]
`;

exports[`Search actions get movies creates FETCH_MOVIES_SUCCESS action when request is successful 1`] = `
Array [
  Object {
    "type": "FETCH_MOVIES_REQUEST",
  },
  Object {
    "payload": Object {
      "movies": Array [
        Object {
          "genres": Array [
            "Test genre",
            "Another test genre",
          ],
          "overview": "Test overview",
          "poster_path": "http://poster.path",
          "release_date": "2020-10-20",
          "runtime": 120,
          "title": "Test tile",
          "vote_average": 9,
        },
        Object {
          "genres": Array [
            "Test genre",
            "Another test genre",
          ],
          "overview": "Test overview",
          "poster_path": "http://poster.path",
          "release_date": "2020-10-20",
          "runtime": 120,
          "title": "Test tile",
          "vote_average": 9,
        },
      ],
      "moviesCount": 2,
    },
    "type": "FETCH_MOVIES_SUCCESS",
  },
]
`;
